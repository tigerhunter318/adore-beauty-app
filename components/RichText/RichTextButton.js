import React from 'react'
import { StyleSheet } from 'react-native'
import Container from '../ui/Container'
import Button from '../ui/Button'
import useUrlNavigation from '../../navigation/utils/useUrlNavigation'

const styles = StyleSheet.create({
  containerStyle: {
    width: '100%'
  },
  textStyle: {
    paddingVertical: 5
  }
})

const RichTextButton = ({ content }) => {
  const urlNavigation = useUrlNavigation()
  const onButtonPress = async () => {
    await urlNavigation.push(content.url)
  }

  return (
    !!content && (
      <Container pv={1} rows justify>
        <Button
          onPress={onButtonPress}
          containerStyle={styles.containerStyle}
          textStyle={styles.textStyle}
          isPrimary={content.type && content.type.toLowerCase() === 'primary'}
        >
          {content.title}
        </Button>
      </Container>
    )
  )
}

export default RichTextButton
