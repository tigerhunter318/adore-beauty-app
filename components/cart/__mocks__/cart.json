{
  "couponIds": [],
  "backordersItems": [],
  "request": {
    "meta": {
      "url": "/ecommerce/payments/braintree/token?cart_id=5fb1184a-8e80-480e-8343-b97a07ec2768",
      "method": {
        "method": "get",
        "headers": {
        }
      }
    },
    "pending": false,
    "data": {
      "data": {
      }
    }
  },
  "checkout": {
    "id": "5fb1184a-8e80-480e-8343-b97a07ec2768",
    "cart": {
      "id": "5fb1184a-8e80-480e-8343-b97a07ec2768",
      "customer_id": 777910,
      "channel_id": 1,
      "email": "noloyalty2@test.com",
      "currency": {
        "code": "AUD"
      },
      "tax_included": true,
      "base_amount": 316.15,
      "discount_amount": 0,
      "cart_amount_inc_tax": 316.15,
      "cart_amount_ex_tax": 287.41,
      "coupons": [],
      "discounts": [
        {
          "id": "fa5b14c5-667a-4d8c-a07c-73c1af0b8bbb",
          "discounted_amount": 0
        },
        {
          "id": "a6289e27-bbbf-4010-ad75-825bc334ecf7",
          "discounted_amount": 0
        },
        {
          "id": "5624a446-0769-46ca-8225-bb725a0097cb",
          "discounted_amount": 0
        },
        {
          "id": "9e68d44c-77d3-5fd9-bd71-1536832d7ab9",
          "discounted_amount": 0
        }
      ],
      "line_items": {
        "physical_items": [
          {
            "id": "fa5b14c5-667a-4d8c-a07c-73c1af0b8bbb",
            "parent_id": null,
            "variant_id": 5170,
            "product_id": 5205,
            "sku": "110716",
            "name": "110716 - Dermalogica MultiVitamin Power Recovery Masque",
            "url": "https://www.adorebeauty.com.au/9664.html",
            "quantity": 1,
            "is_taxable": true,
            "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/5205/images/100031/png_multivitamin_power_recovery_masque__06607.1604377185.220.290.png?c=1",
            "discounts": [],
            "coupons": [],
            "discount_amount": 0,
            "coupon_amount": 0,
            "original_price": 93.25,
            "list_price": 93.25,
            "sale_price": 93.25,
            "extended_list_price": 93.25,
            "extended_sale_price": 93.25,
            "is_require_shipping": true,
            "gift_wrapping": null,
            "is_mutable": true
          },
          {
            "id": "a6289e27-bbbf-4010-ad75-825bc334ecf7",
            "parent_id": null,
            "variant_id": 6718,
            "product_id": 6753,
            "sku": "111279",
            "name": "111279 - Dermalogica Sound Sleep Cocoon Transformative Night Gel-Cream",
            "url": "https://www.adorebeauty.com.au/703070.html",
            "quantity": 1,
            "is_taxable": true,
            "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/6753/images/116541/sound_sleep_cocoon__83705.1604385914.220.290.png?c=1",
            "discounts": [],
            "coupons": [],
            "discount_amount": 0,
            "coupon_amount": 0,
            "original_price": 129.8,
            "list_price": 129.8,
            "sale_price": 129.8,
            "extended_list_price": 129.8,
            "extended_sale_price": 129.8,
            "is_require_shipping": true,
            "gift_wrapping": null,
            "is_mutable": true
          },
          {
            "id": "5624a446-0769-46ca-8225-bb725a0097cb",
            "parent_id": null,
            "variant_id": 27939,
            "product_id": 38636,
            "sku": "111453",
            "name": "111453 - Dermalogica Daily Milkfoliant",
            "url": "https://www.adorebeauty.com.au/1190062.html",
            "quantity": 1,
            "is_taxable": true,
            "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/38636/images/152363/Dermalogica_Daily_Milkfoliant_hero__19133.1654848620.220.290.png?c=1",
            "discounts": [],
            "coupons": [],
            "discount_amount": 0,
            "coupon_amount": 0,
            "original_price": 93.1,
            "list_price": 93.1,
            "sale_price": 93.1,
            "extended_list_price": 93.1,
            "extended_sale_price": 93.1,
            "is_require_shipping": true,
            "gift_wrapping": null,
            "is_mutable": true
          },
          {
            "id": "9e68d44c-77d3-5fd9-bd71-1536832d7ab9",
            "parent_id": null,
            "variant_id": 14457,
            "product_id": 14492,
            "sku": "2-free-samples",
            "name": "2-free-samples - 2 Free Samples",
            "url": "https://www.adorebeauty.com.au/957669.html",
            "quantity": 1,
            "is_taxable": true,
            "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/14492/images/95499/samples_png_1__54490.1604375578.220.290.jpg?c=1",
            "discounts": [
              {
                "id": 20,
                "discounted_amount": 0
              }
            ],
            "coupons": [],
            "discount_amount": 0,
            "coupon_amount": 0,
            "original_price": 0,
            "list_price": 0,
            "sale_price": 0,
            "extended_list_price": 0,
            "extended_sale_price": 0,
            "is_require_shipping": true,
            "gift_wrapping": null,
            "is_mutable": false
          }
        ],
        "digital_items": [],
        "gift_certificates": [],
        "custom_items": []
      },
      "created_time": "2023-05-25T05:32:18+00:00",
      "updated_time": "2023-05-26T00:16:57+00:00"
    },
    "billing_address": {
      "id": "646ef2e221fbd",
      "first_name": "Sad",
      "last_name": "Dsa",
      "email": "noloyalty2@test.com",
      "company": "",
      "address1": "44A Maddox Road",
      "address2": "",
      "city": "NEWPORT",
      "state_or_province": "Victoria",
      "state_or_province_code": "VIC",
      "country": "Australia",
      "country_code": "AU",
      "postal_code": "3015",
      "phone": "0400000000",
      "custom_fields": []
    },
    "consignments": [
      {
        "id": "646f37540c908",
        "shipping_cost_inc_tax": 9.95,
        "shipping_cost_ex_tax": 9.05,
        "handling_cost_inc_tax": 0,
        "handling_cost_ex_tax": 0,
        "coupon_discounts": [],
        "discounts": [],
        "line_item_ids": [
          "fa5b14c5-667a-4d8c-a07c-73c1af0b8bbb",
          "a6289e27-bbbf-4010-ad75-825bc334ecf7",
          "5624a446-0769-46ca-8225-bb725a0097cb",
          "9e68d44c-77d3-5fd9-bd71-1536832d7ab9"
        ],
        "selected_shipping_option": {
          "id": "49f7275637505d02bc5fa5b18ee0af3f",
          "type": "sameday",
          "description": "Australia Post - TWILIGHT (Tomorrow  6PM - 10PM)",
          "image_url": "",
          "cost": 9.95,
          "transit_time": "",
          "additional_description": ""
        },
        "shipping_address": {
          "first_name": "Sad",
          "last_name": "Dsa",
          "email": "noloyalty2@test.com",
          "company": "",
          "address1": "44A Maddox Road",
          "address2": "",
          "city": "NEWPORT",
          "state_or_province": "Victoria",
          "state_or_province_code": "VIC",
          "country": "Australia",
          "country_code": "AU",
          "postal_code": "3015",
          "phone": "0400000000",
          "custom_fields": []
        },
        "address": {
          "first_name": "Sad",
          "last_name": "Dsa",
          "email": "noloyalty2@test.com",
          "company": "",
          "address1": "44A Maddox Road",
          "address2": "",
          "city": "NEWPORT",
          "state_or_province": "Victoria",
          "state_or_province_code": "VIC",
          "country": "Australia",
          "country_code": "AU",
          "postal_code": "3015",
          "phone": "0400000000",
          "custom_fields": []
        },
        "available_shipping_options": [
          {
            "id": "3c92924a7bf41fb0c3cc389e0f4ba9c9",
            "type": "flat",
            "description": "Adore Society Australia - Express Post - 1-3 day (Express Post)",
            "image_url": "",
            "cost": 0,
            "transit_time": "",
            "additional_description": ""
          },
          {
            "id": "c44c73b389d3c8df4d27bd0017510983",
            "type": "sameday",
            "description": "Australia Post - TWILIGHT (Tonight 6PM - 10PM)",
            "image_url": "",
            "cost": 9.95,
            "transit_time": "",
            "additional_description": ""
          }
        ]
      }
    ],
    "taxes": [
      {
        "name": "GST",
        "amount": 28.74
      }
    ],
    "coupons": [],
    "order_id": null,
    "shipping_cost_total_inc_tax": 0,
    "shipping_cost_total_ex_tax": 0,
    "handling_cost_total_inc_tax": 0,
    "handling_cost_total_ex_tax": 0,
    "tax_total": 28.74,
    "subtotal_inc_tax": 316.15,
    "subtotal_ex_tax": 287.41,
    "grand_total": 316.15,
    "created_time": "2023-05-25T05:32:18+00:00",
    "updated_time": "2023-05-26T00:16:57+00:00",
    "customer_message": ""
  },
  "cartItemsProductDetail": [
    {
      "__typename": "Products",
      "product_id": 1190062,
      "name": "Dermalogica Daily Milkfoliant",
      "price": "93.1",
      "productImage": "https://www.adorebeauty.com.au/pim_media/000/409/339/Dermalogica_Daily_Milkfoliant_hero.png",
      "category_name": [
        "Skin Care",
        "Cosmeceuticals",
        "Dermalogica",
        "Exfoliants",
        "Skincare Concerns",
        "Skincare for Oily Skin",
        "Skincare for Dry Skin",
        "Skincare for Combination Skin",
        "Skincare for Normal Skin",
        "Cleansers & Scrubs",
        "Face Scrubs & Exfoliating Cleansers",
        "Serums & Treatments",
        "Peels and Exfoliating Treatments",
        "AHAs"
      ],
      "size": null,
      "brand_name": "Dermalogica",
      "productSku": [
        "111453"
      ],
      "product_url": "https://www.adorebeauty.com.au/p/dermalogica/dermalogica-daily-milkfoliant.html",
      "inStock": 1
    },
    {
      "__typename": "Products",
      "product_id": 703070,
      "name": "Dermalogica Sound Sleep Cocoon Transformative Night Gel-Cream",
      "price": "129.8",
      "productImage": "https://www.adorebeauty.com.au/media/catalog/product/s/o/sound_sleep_cocoon.png",
      "category_name": [
        "Skin Care",
        "Cosmeceuticals",
        "Dermalogica",
        "Moisturisers",
        "Moisturisers",
        "Night Cream",
        "Serums & Treatments"
      ],
      "size": "50ml",
      "brand_name": "Dermalogica",
      "productSku": [
        "111279"
      ],
      "product_url": "https://www.adorebeauty.com.au/p/dermalogica/dermalogica-transformative-night-gel-cream.html",
      "inStock": 1
    },
    {
      "__typename": "Products",
      "product_id": 9664,
      "name": "Dermalogica MultiVitamin Power Recovery Masque",
      "price": "93.25",
      "productImage": "https://www.adorebeauty.com.au/media/catalog/product/p/n/png_multivitamin_power_recovery_masque.png",
      "category_name": [
        "Skin Care",
        "Cosmeceuticals",
        "Dermalogica",
        "AGE Smart",
        "Masques",
        "Skincare Concerns",
        "Anti-Ageing Skincare",
        "Skincare for Combination Skin",
        "Skincare for Normal Skin",
        "Anti-Pollution",
        "Pigmentation Correction",
        "Masks",
        "Vitamin A/Retinol",
        "Vitamin C",
        "Vitamin B",
        "Vitamin B Complexes",
        "AHAs"
      ],
      "size": "75",
      "brand_name": "Dermalogica",
      "productSku": [
        "110716"
      ],
      "product_url": "https://www.adorebeauty.com.au/p/dermalogica/dermalogica-multivitamin-power-recovery-masque.html",
      "inStock": 1
    },
    {
      "__typename": "Products",
      "product_id": 957669,
      "name": "2 Free Samples",
      "price": "0",
      "productImage": "https://www.adorebeauty.com.au/media/catalog/product/s/a/samples_png_1.png",
      "category_name": null,
      "size": null,
      "brand_name": "Gift",
      "productSku": [
        "2-free-samples"
      ],
      "product_url": "https://www.adorebeauty.com.au/p/gift/2-free-samples.html",
      "inStock": 1
    }
  ],
  "billingAddress": {
    "addressMeta": {
      "id": 16038328,
      "addressable_id": 1080330,
      "big_commerce_id": 1001238,
      "company": null,
      "street_1": "44A Maddox Road",
      "street_2": null,
      "city": "NEWPORT",
      "country": "Australia",
      "state": "VIC",
      "post_code": "3015",
      "phone": "0400000000",
      "is_default_shipping": false,
      "is_default_billing": false,
      "created_at": "2023-05-25T10:13:11.000000Z",
      "updated_at": "2023-05-25T10:16:12.000000Z",
      "type": "shipping",
      "addressable_type": "App\\Models\\Customer",
      "external_id": null,
      "first_name": "Sad",
      "last_name": "Dsa",
      "address_line_1": "44A Maddox Road",
      "address_line_2": null,
      "locality_name": "NEWPORT",
      "state_territory": "VIC",
      "postcode": "3015"
    },
    "address": "44A Maddox Road, NEWPORT, VIC, 3015",
    "phone": "0400000000",
    "firstName": "Sad",
    "lastName": "Dsa",
    "newAddress": "",
    "isSaveAddressChecked": false
  },
  "orderConfirmation": null,
  "addedToCartItems": [],
  "promotionItems": [],
  "promotions": [],
  "paymentMethod": null,
  "newBillingAddress": false,
  "cartItemsInventory": [
    {
      "id": "fa5b14c5-667a-4d8c-a07c-73c1af0b8bbb",
      "parent_id": null,
      "variant_id": 5170,
      "product_id": 5205,
      "sku": "110716",
      "name": "110716 - Dermalogica MultiVitamin Power Recovery Masque",
      "url": "https://www.adorebeauty.com.au/9664.html",
      "quantity": 1,
      "is_taxable": true,
      "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/5205/images/100031/png_multivitamin_power_recovery_masque__06607.1604377185.220.290.png?c=1",
      "discounts": [],
      "coupons": [],
      "discount_amount": 0,
      "coupon_amount": 0,
      "original_price": 93.25,
      "list_price": 93.25,
      "sale_price": 93.25,
      "extended_list_price": 93.25,
      "extended_sale_price": 93.25,
      "is_require_shipping": true,
      "gift_wrapping": null,
      "is_mutable": true,
      "availability": "available",
      "inventory_level": 127
    },
    {
      "id": "a6289e27-bbbf-4010-ad75-825bc334ecf7",
      "parent_id": null,
      "variant_id": 6718,
      "product_id": 6753,
      "sku": "111279",
      "name": "111279 - Dermalogica Sound Sleep Cocoon Transformative Night Gel-Cream",
      "url": "https://www.adorebeauty.com.au/703070.html",
      "quantity": 1,
      "is_taxable": true,
      "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/6753/images/116541/sound_sleep_cocoon__83705.1604385914.220.290.png?c=1",
      "discounts": [],
      "coupons": [],
      "discount_amount": 0,
      "coupon_amount": 0,
      "original_price": 129.8,
      "list_price": 129.8,
      "sale_price": 129.8,
      "extended_list_price": 129.8,
      "extended_sale_price": 129.8,
      "is_require_shipping": true,
      "gift_wrapping": null,
      "is_mutable": true,
      "availability": "available",
      "inventory_level": 196
    },
    {
      "id": "5624a446-0769-46ca-8225-bb725a0097cb",
      "parent_id": null,
      "variant_id": 27939,
      "product_id": 38636,
      "sku": "111453",
      "name": "111453 - Dermalogica Daily Milkfoliant",
      "url": "https://www.adorebeauty.com.au/1190062.html",
      "quantity": 1,
      "is_taxable": true,
      "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/38636/images/152363/Dermalogica_Daily_Milkfoliant_hero__19133.1654848620.220.290.png?c=1",
      "discounts": [],
      "coupons": [],
      "discount_amount": 0,
      "coupon_amount": 0,
      "original_price": 93.1,
      "list_price": 93.1,
      "sale_price": 93.1,
      "extended_list_price": 93.1,
      "extended_sale_price": 93.1,
      "is_require_shipping": true,
      "gift_wrapping": null,
      "is_mutable": true,
      "availability": "available",
      "inventory_level": 167
    },
    {
      "id": "9e68d44c-77d3-5fd9-bd71-1536832d7ab9",
      "parent_id": null,
      "variant_id": 14457,
      "product_id": 14492,
      "sku": "2-free-samples",
      "name": "2-free-samples - 2 Free Samples",
      "url": "https://www.adorebeauty.com.au/957669.html",
      "quantity": 1,
      "is_taxable": true,
      "image_url": "https://cdn11.bigcommerce.com/s-61yxa1x1fw/products/14492/images/95499/samples_png_1__54490.1604375578.220.290.jpg?c=1",
      "discounts": [
        {
          "id": 20,
          "discounted_amount": 0
        }
      ],
      "coupons": [],
      "discount_amount": 0,
      "coupon_amount": 0,
      "original_price": 0,
      "list_price": 0,
      "sale_price": 0,
      "extended_list_price": 0,
      "extended_sale_price": 0,
      "is_require_shipping": true,
      "gift_wrapping": null,
      "is_mutable": false,
      "availability": "available",
      "inventory_level": 95975
    }
  ],
  "productPending": null,
  "giftCertificates": [],
  "recentlyOrderedProducts": {},
  "storeCredits": 0,
  "paymentTypes": {},
  "paymentCardDetails": {},
  "customerOrders": {},
  "checkoutOrder": null
}
