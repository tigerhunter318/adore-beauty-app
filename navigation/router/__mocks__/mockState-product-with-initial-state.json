{
  "stale": false,
  "type": "stack",
  "key": "stack-d_Rvaheu02xudocRnM-EZ",
  "index": 0,
  "routeNames": [
    "MainDrawer",
    "Login",
    "ProductQuickView",
    "HasuraPromoQuickView",
    "PromoQuickView",
    "ArticleShopProducts",
    "DeeplinkLoader",
    "AdoreSocietyModalScreen",
    "CMS",
    "NoInternetConnection"
  ],
  "routes": [
    {
      "key": "MainDrawer-qcdvNKHEDZ7OTgPbjkAI9",
      "name": "MainDrawer",
      "state": {
        "stale": false,
        "type": "drawer",
        "key": "drawer-ttg2qAkDtUciWJJAa7Kq8",
        "index": 0,
        "routeNames": [
          "Main"
        ],
        "history": [
          {
            "type": "route",
            "key": "Main-uBG3x92W3K7h3b-d819pm"
          }
        ],
        "routes": [
          {
            "name": "Main",
            "key": "Main-uBG3x92W3K7h3b-d819pm",
            "state": {
              "stale": false,
              "type": "stack",
              "key": "stack-aYsvtsb8JMnXdwECagzt9",
              "index": 1,
              "routeNames": [
                "MainTab",
                "Cart",
                "ProductStack",
                "GiftCertificate",
                "PostScreen",
                "BeautyIQPodcastEpisode",
                "OrderConfirm"
              ],
              "routes": [
                {
                  "key": "MainTab-BlVP17sWVFtGPi-QS-5KX",
                  "name": "MainTab",
                  "params": {
                    "initial": true,
                    "screen": "Shop",
                    "params": {
                      "initial": true,
                      "screen": "ShopCategoryProducts",
                      "params": {
                        "fromScreenPath": "/MainDrawer/Main/ProductStack?initial=true&params=%5Bobject%20Object%5D&screen=Product",
                        "showLoader": true,
                        "isDeepLink": true,
                        "url": "make-up",
                        "name": ""
                      }
                    }
                  },
                  "state": {
                    "stale": false,
                    "type": "tab",
                    "key": "tab-b2XMG1psmaTyt_QJunbV7",
                    "index": 1,
                    "routeNames": [
                      "Home",
                      "Shop",
                      "CartTab",
                      "Account",
                      "BeautyIQ"
                    ],
                    "history": [
                      {
                        "type": "route",
                        "key": "Home-hk8ll7oNJQwAU99t7mM8I"
                      },
                      {
                        "type": "route",
                        "key": "Shop-NTmuZzGOswQy_Qu7lWYBV"
                      }
                    ],
                    "routes": [
                      {
                        "name": "Home",
                        "key": "Home-hk8ll7oNJQwAU99t7mM8I",
                        "state": {
                          "stale": false,
                          "type": "stack",
                          "key": "stack-mvSosZ6jha7Kb66oOFvr5",
                          "index": 0,
                          "routeNames": [
                            "Home"
                          ],
                          "routes": [
                            {
                              "key": "Home-RxPeC6vASDcCRo3rJOh3H",
                              "name": "Home",
                              "params": {
                                "screenHeaderLogoSource": "https://images.ctfassets.net/pi28xy1s107o/52yUOD2lVWND8WFiTSi0ZL/140510078dd7784734f219dd16fee9be/ab-app-logo-christmas-rust-colour.png"
                              }
                            }
                          ]
                        }
                      },
                      {
                        "name": "Shop",
                        "key": "Shop-NTmuZzGOswQy_Qu7lWYBV",
                        "params": {
                          "initial": true,
                          "screen": "ShopCategoryProducts",
                          "params": {
                            "fromScreenPath": "/MainDrawer/Main/ProductStack?initial=true&params=%5Bobject%20Object%5D&screen=Product",
                            "showLoader": true,
                            "isDeepLink": true,
                            "url": "make-up",
                            "name": ""
                          },
                          "fromScreenPath": "/MainDrawer/Main/MainTab/Shop/ShopCategoryProducts?fromScreenPath=%2FMainDrawer%2FMain%2FMainTab%2FHome%2FHome%3FscreenHeaderLogoSource%3Dhttps%253A%252F%252Fimages.ctfassets.net%252Fpi28xy1s107o%252F52yUOD2lVWND8WFiTSi0ZL%252F140510078dd7784734f219dd16fee9be%252Fab-app-logo-christmas-rust-colour.png&isDeepLink=true&name=Night%20Cream&showLoader=true&url=skin-care%2Fmoisturisers%2Fnight",
                          "showLoader": true,
                          "isDeepLink": true
                        },
                        "state": {
                          "stale": false,
                          "type": "stack",
                          "key": "stack--YFUTNEnB8hmxLubCqokx",
                          "index": 1,
                          "routeNames": [
                            "Shop",
                            "ShopCategoryProducts",
                            "ShopTabs",
                            "ShopPromotions"
                          ],
                          "routes": [
                            {
                              "key": "Shop-EzEUbC3oDCUIXSTHhMS4s",
                              "name": "Shop",
                              "params": {
                                "fromScreenPath": "/MainDrawer/Main/MainTab/Shop/ShopCategoryProducts?fromScreenPath=%2FMainDrawer%2FMain%2FMainTab%2FHome%2FHome%3FscreenHeaderLogoSource%3Dhttps%253A%252F%252Fimages.ctfassets.net%252Fpi28xy1s107o%252F52yUOD2lVWND8WFiTSi0ZL%252F140510078dd7784734f219dd16fee9be%252Fab-app-logo-christmas-rust-colour.png&isDeepLink=true&name=Night%20Cream&showLoader=true&url=skin-care%2Fmoisturisers%2Fnight",
                                "showLoader": true,
                                "isDeepLink": true
                              }
                            },
                            {
                              "key": "ShopCategoryProducts-y7o-kdbSSlevCDIt5N-wh",
                              "name": "ShopCategoryProducts",
                              "params": {
                                "fromScreenPath": "/MainDrawer/Main/ProductStack?initial=true&params=%5Bobject%20Object%5D&screen=Product",
                                "showLoader": true,
                                "isDeepLink": true,
                                "url": "make-up",
                                "name": "Makeup"
                              }
                            }
                          ]
                        }
                      },
                      {
                        "name": "CartTab",
                        "key": "CartTab-zRnK01ka6OjAYp3IfYjIe"
                      },
                      {
                        "name": "Account",
                        "key": "Account-HgaUqlSE3VpFuxJ6KqCDd"
                      },
                      {
                        "name": "BeautyIQ",
                        "key": "BeautyIQ-3QJ3M1_VaGFwfF5VbhNNX"
                      }
                    ]
                  }
                },
                {
                  "key": "ProductStack-yjKOvF2B0sb4BZ827njFD",
                  "name": "ProductStack",
                  "params": {
                    "initial": true,
                    "screen": "Product",
                    "params": {
                      "fromScreenPath": "/MainDrawer/Main/MainTab/Shop/ShopCategoryProducts?fromScreenPath=%2FMainDrawer%2FMain%2FProductStack%3Finitial%3Dtrue%26params%3D%255Bobject%2520Object%255D%26screen%3DProduct&isDeepLink=true&name=Makeup&showLoader=true&url=make-up",
                      "showLoader": true,
                      "isDeepLink": true,
                      "identifier": "the-ordinary-hyaluronic-acid-2-b5-60ml"
                    }
                  }
                }
              ]
            }
          }
        ]
      }
    }
  ]
}
