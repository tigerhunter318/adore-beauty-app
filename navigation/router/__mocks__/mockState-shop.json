{
  "stale": false,
  "type": "stack",
  "key": "stack-ND4CpSjpTnb-1C3h4gs2h",
  "index": 0,
  "routeNames": [
    "MainDrawer",
    "Login",
    "ProductQuickView",
    "HasuraPromoQuickView",
    "PromoQuickView",
    "ArticleShopProducts",
    "DeeplinkLoader",
    "AdoreSocietyModalScreen",
    "CMS",
    "NoInternetConnection"
  ],
  "routes": [
    {
      "key": "MainDrawer-34orCYFcY9niQFI_evSQS",
      "name": "MainDrawer",
      "state": {
        "stale": false,
        "type": "drawer",
        "key": "drawer-cJnUbTewQUImrQEnIBwQI",
        "index": 0,
        "routeNames": [
          "Main"
        ],
        "history": [
          {
            "type": "route",
            "key": "Main-FkOPQVKNv1TVWc4cY_TKX"
          }
        ],
        "routes": [
          {
            "name": "Main",
            "key": "Main-FkOPQVKNv1TVWc4cY_TKX",
            "state": {
              "stale": false,
              "type": "stack",
              "key": "stack-JbYzuh7ntGhF8JvF4OMTj",
              "index": 0,
              "routeNames": [
                "MainTab",
                "Cart",
                "ProductStack",
                "GiftCertificate",
                "PostScreen",
                "BeautyIQPodcastEpisode",
                "OrderConfirm"
              ],
              "routes": [
                {
                  "key": "MainTab-odFB8UsmP9gftXngI-XiC",
                  "name": "MainTab",
                  "params": {
                    "initial": true,
                    "screen": "Home",
                    "params": {
                      "screenHeaderLogoSource": "",
                      "showLoader": true,
                      "isDeepLink": true
                    }
                  },
                  "state": {
                    "stale": false,
                    "type": "tab",
                    "key": "tab-WJuXdwSZG8qRyx_UUdfX_",
                    "index": 1,
                    "routeNames": [
                      "Home",
                      "Shop",
                      "CartTab",
                      "Account",
                      "BeautyIQ"
                    ],
                    "history": [
                      {
                        "type": "route",
                        "key": "Home-vV8xfWV_1GqXAc22EP4f4"
                      },
                      {
                        "type": "route",
                        "key": "Shop-JhjYbGnGKxk8M-5rYsUmM"
                      }
                    ],
                    "routes": [
                      {
                        "name": "Home",
                        "key": "Home-vV8xfWV_1GqXAc22EP4f4",
                        "params": {
                          "screenHeaderLogoSource": "",
                          "showLoader": true,
                          "isDeepLink": true
                        },
                        "state": {
                          "stale": false,
                          "type": "stack",
                          "key": "stack-gBYCx8kB6ErIdao2VoPVt",
                          "index": 0,
                          "routeNames": [
                            "Home"
                          ],
                          "routes": [
                            {
                              "key": "Home-ROu2hL5yOT18zEiciSo59",
                              "name": "Home",
                              "params": {
                                "screenHeaderLogoSource": "https://images.ctfassets.net/pi28xy1s107o/52yUOD2lVWND8WFiTSi0ZL/140510078dd7784734f219dd16fee9be/ab-app-logo-christmas-rust-colour.png"
                              }
                            }
                          ]
                        }
                      },
                      {
                        "name": "Shop",
                        "key": "Shop-JhjYbGnGKxk8M-5rYsUmM"
                      },
                      {
                        "name": "CartTab",
                        "key": "CartTab-D60pLbtMgi0T78KKbT9uV"
                      },
                      {
                        "name": "Account",
                        "key": "Account-S2fhVqdf3aULrZ-affjXm"
                      },
                      {
                        "name": "BeautyIQ",
                        "key": "BeautyIQ-nMTJBYgk9GX0ax737nRpu"
                      }
                    ]
                  }
                }
              ]
            }
          }
        ]
      }
    }
  ]
}
