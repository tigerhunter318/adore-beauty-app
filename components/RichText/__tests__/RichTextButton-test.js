import React from 'react'
import renderer from 'react-test-renderer'
import { getArticleContent } from './__mocks__/helper'
import RichTextButton from '../RichTextButton'

it(`RichTextButton renders for alterna-air-dry-balm-review`, () => {
  const parsedContent = getArticleContent('button', 'alterna-air-dry-balm-review')
  const tree = renderer.create(<RichTextButton content={parsedContent} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it(`RichTextButton renders for what-is-a-serum`, () => {
  const parsedContent = getArticleContent('button', 'what-is-a-serum')
  const tree = renderer.create(<RichTextButton content={parsedContent} />).toJSON()
  expect(tree).toMatchSnapshot()
})
