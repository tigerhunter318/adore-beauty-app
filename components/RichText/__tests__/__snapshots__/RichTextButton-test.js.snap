// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextButton renders for alterna-air-dry-balm-review 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      Object {},
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Regular",
          },
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            undefined,
            Object {},
            Object {
              "color": "black",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 14,
            },
            false,
            false,
            false,
            false,
            Object {
              "paddingVertical": 5,
            },
          ],
        ]
      }
    >
      Shop Alterna Hair My Hair My Canvas Easy Does It Air-Dry Balm
    </Text>
  </View>
</View>
`;

exports[`RichTextButton renders for what-is-a-serum 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      Object {},
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-Regular",
          },
          Array [
            Object {
              "color": "white",
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            undefined,
            Object {},
            Object {
              "color": "black",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 14,
            },
            false,
            false,
            false,
            false,
            Object {
              "paddingVertical": 5,
            },
          ],
        ]
      }
    >
      Shop Revolution Skincare 12.5% Vitamin C Serum Here
    </Text>
  </View>
</View>
`;
